### PR TITLE
Do not warn on nested route errors

### PR DESCRIPTION
**Pull request by ryardley**
Closes: #597 

### What are the changes and their implications?

* Do not warn on nested route errors
* Sibling Api folders are still compiled to pages/api like before
* Added missing unit test on duplicate route function
* Added unit test on entire page stage at the stream level
* Fixed bug in displaying code that was displaying too many matches.

### Checklist

- [x] Tests added for changes
- [x] User facing changes documented

### CODE REVIEW

**Comment by ryardley** [comment]
There is contention here in #597 so we can’t merge this. Good to have eyes on it.

**Comment by wKovacs64** [comment]
Yep, I'm following that. Just fulfilling my review request in the event you convince Brandon. 😉

**Comment by ryardley** [comment]
Ok adjusted the PR based on #597 to remove the Nested Route check. Please review and I will merge.

**Comment by ryardley** [comment]
ooh @flybayer - you know what I am not sure maybe I will add a test case for that here.

**Comment by ryardley** [comment]
Extra testing has been added. Found a bug where we were displaying all the routes multiple times depending on how many matches we were getting so I rewrote the duplication checking logic and added appropriate tests. Just checking all the windows stuff works then should be ready to merge.

File: packages/server/src/stages/pages/index.ts [hunk 1]
```diff
@@ -17,11 +78,14 @@ export const createStagePages: Stage = ({config, bus, getInputCache}) => {
...
 
-  const stream = through.obj((file: File, _, next) => {
+  const stream: NodeJS.ReadWriteStream = through.obj((file: File, _, next) => {
     const entries = getInputCache().toPaths()
 
+    const duplicates = findDuplicates(entries, fullTransformer)
+
     // Check for duplicate pages entries
-    const duplicatePages = getDuplicatePaths(entries, 'pages')
+    const duplicatePages = filterBy(duplicates, 'pages', 'api')
```
Comment (merelinguist) [inline packages/server/src/stages/pages/index.ts:87]: Nice refactoring!
